akd_core/src/types/mod.rs

```diff
@@ -44,11 +46,40 @@ pub trait SizeOf {
 /// This type is used to indicate a direction for a
 /// particular node relative to its parent. We use
 /// 0 to represent "left" and 1 to represent "right".
-pub type Direction = Option<usize>;
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+#[cfg_attr(
+    feature = "serde_serialization",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+#[repr(u8)]
+pub enum Direction {
+    /// Left
+    Left = 0u8,
```

Comment:
Perhaps a style thing, but we don't need to specify 0u8, 1u8, 255u8 when we have `#[repr(u8)]` right? Also, perhaps we should use u8::MAX instead of 255 everywhere, or vice versa?

Reply:
We do, since by default it would encode 0, 1, 2. In this case, since we're going to encode over the wire I'd like it clear that "None" is really a separate value and not leave it up to the compiler to infer (perhaps incorrectly).

This just removes any risk in case someone re-orders enum fields or something. To the `u8::MAX` point, good catch will adjust!

Reply:
Sorry I meant, do we need to say `0u8`, `1u8`, `255u8` instead of just `0`, `1`, `255` / `u8::MAX`?